Wrappers/Python/cil/optimisation/operators/Operator.py

```diff
@@ -217,28 +218,36 @@ def PowerMethod(operator, max_iteration=10, initial=None, tolerance = 1e-5,  ret
         i = 0
         while (i < max_iteration and diff > tolerance):
             i+=1
-            
+            if i==max_iteration:

```

Comment:
This check can move outside the loop and we run once at the end as `i` is still in scrope.

However I'm not sure about this warning.  We  typically  run 10 iterations when we call it through `calculate_norm`. And I don't know if this usually brings us inside the tolerance or not. Potentially we could allow the tolerance to be set to `False` or `None` and not show the warning.

Reply:
That's a good point. Perhaps instead I could just make a note in the documentation? Other options could be a check_convergence flag which produces an error if it doesn't converge or an additional return of True or False if it converges when return_all is flagged?

Reply:
Added a convergence flag in the case return_all=True instead of returning this warning